openpype/hosts/tvpaint/plugins/publish/collect_instance_frames.py

```diff
@@ -34,3 +34,103 @@ def process(self, instance):
+                # Remove frames lower that the tracker frameStart
+                custom_frames = [frame for frame in custom_frames if (start_value < frame)]
+                # Replace by the true tracker frameStart
+                custom_frames.insert(0, start_value)
+
+            instance.data["customFrames"] = custom_frames
+
+            # Update the instance data
+            instance.data["frameStart"] = min(custom_frames)
+            instance.data["frameEnd"] = max(custom_frames)
+
+            if custom_instance_frames:
+                self.log.info("Export Custom frames {}".format(custom_frames))
+
+        if custom_instance_frames or keep_frame_index:
+            self.log.info("Changed frames Start/End {}-{} on instance {} ".format(instance.data["frameStart"] , instance.data["frameEnd"], instance.data["subset"]))
+
+    def list_custom_frames_export(self, custom_frames , mark_in, mark_out, sceneStartFrame):
+        """
+        Create a list of frame to export based on a string
+
+        Args:
+            custom_frames(str): frames to export, can be :
+                                "1, 4, 6"
+                                "[1-6], 15"
+                                "[:-4], 6"
+                                "1, 4, [6-:]"
+                                the ":" implise that it will go to the mark_in or to the mark_out
+            mark_in(int): frame on which is set markIn in tvpp
+            mark_out(int): frame on which is set markOut in tvpp
+            sceneStartFrame(int): frame de start du projet tvpp
+
+        Returns:
+            list: A interpreted list of int based on the str input
+        """
+        # if no str is given, return a range based on mark_in and mark_out
+        if not custom_frames:
+            return list(range(int(mark_in), int(mark_out) + 1))
+
+        # Search intervals with regex
+        pattern = r'\[(\d+|:)(?:-(\d+|:))?\]'
+        range_pattern = re.findall(pattern, custom_frames)
+
+        # Replace all intervals by int range
+        for start, end in range_pattern:
+            # Process Start of interval
+            expanded_start = start
+            if start in (':'):
+                # Security if no start defined or
+                # Case if must start on mark_in
+                expanded_start = mark_in + sceneStartFrame
+
+            # Process End of interval
+            expanded_end = end
+            if end in (":"):
+                # Security if no end defined or
+                # Case if must end on mark_out
+                expanded_end = mark_out + sceneStartFrame
+
+            # check if the end if AFTER the start
+            # Can happen if the user set the markIn AFTER the end frame he entered in the custom frame string
+            if int(expanded_end) < int(expanded_start):
+                self.log.warning("The End frame in [:-{}] is lower than the tvpp markIn {}".format(expanded_end, start))
```

Comment:
We shouldn't encounter this case, if we want only one frame, instaed of [10-10] we just should enter 10.

I tested this configuration: 
- In tvpp put the markIn à 27
- custom_frames = "[:-27]"

It will just return [27]

Reply:
Ne jamais faire confiances aux bon sens des utilisateurs